rust/datafusion/src/logical_plan/expr.rs

```diff
@@ -149,6 +149,13 @@ pub enum Expr {
         /// The `DataType` the expression will yield
         data_type: DataType,
     },
+    /// Casts the expression to a given type and returns null if not possible. This expression is guaranteed to have a fixed type.
```

Comment:
it may be good to add a note to `Cast` saying that a runtime error will result if the expression can not be cast.